### DISCUSSION TITLE
10.15 - 폴더 구조와 이미지

### DISCUSSION THREAD

**Discussion by ghost** [team-mokcha/corabal-beta#37] [주간 회의록]
1. services
- tsx -> ts
- 하위 폴더를 만들어서 폴더(functions) 안에다가 밀어넣기
- ...services.ts에서 services 빼기
- 폴더구조
```
services
  |-firebaseApp.ts
  |-keys.ts
  |-functions
    |-handle-xxx.ts (대쉬 넣을지 말지?)
```
2. assets
현재 문제
- assets 폴더에 다 있으므로 번들 사이즈가 커짐 
- 클라이언트 사이드에서 렌더링을 해서 보여주게 된다면 로딩 속도가 저하될 것임
솔루션
- 이미지는 assets 폴더에 있는 것보다 url로 불러와서 쓰는 방법이 괜찮을 것 같음 (firestore에 컵 이미지 박은 것처럼) 
  => 이 방식을 사용할 거라면 한 번 알아봐야 함! "속도" 면에 중점을 두고 알아봐야 함 
  => 구글 드라이브도 괜찮을 것 같음 (퍼포먼스가 어떻게 나올지 몰라서 알아봐야 할 것임)
- 폴더 구조 나누는 건 common / view 이름으로
```
assets
  |- common
    |- btn_x.png
  |- index
    |- btn_y.png
```

3. src
- 컴포넌트들이 최소 단위(gradient button 등)인 것들과 뷰에서 나눠져 나오는 것(min - carousel)들을 어떻게 분리할지 찾아봐야 함
- css 파일을 style.module.css 로 하면 충돌 안 남! 근데 지금 우리 파일 구조의 css도 충돌은 안 나니까 이것도 베스트 프랙티스 알아보기
- style, styles 둘 중에 이름 통일하기 => 엑스포 문서 보고 코딩 컨벤션을 맞는 방식으로 통일해서 사용하기

4. 조사해올 것
- 폴더 구조 어떻게 할지 -> 수연
- 이미지 url로 불러올 때 어떤 플랫폼을 사용할지 -> 혜진

**Comment by HyejinYang** [comment]
assets 이슈 - 다시 생각해보니, 오프라인 상태에서도 이미지 볼 수 있게 하려면 지금 방식이 맞음
